components/director/internal/domain/labeldef/service_test.go

```diff
@@ -140,19 +141,232 @@ func TestServiceList(t *testing.T) {
+			ID:     fixUID(),
+			Tenant: tenant,
+			Key:    key,
+			Schema: fixBasicSchema(t),
+		}
+
+		in := model.LabelDefinition{
+			ID:     fixUID(),
+			Key:    key,
+			Tenant: tenant,
+			Schema: newSchema,
+		}
+
+		expectedLD := model.LabelDefinition{
+			ID:     fixUID(),
+			Tenant: tenant,
+			Key:    key,
+			Schema: newSchema,
+		}
+
+		existingLabels := []*model.Label{
+			{
+				ID:         "b9566e9d-83a2-4091-8c65-7a512b88f89e",
+				Tenant:     tenant,
+				Key:        key,
+				Value:      "val",
+				ObjectID:   "foo",
+				ObjectType: model.RuntimeLabelableObject,
+			},
+			{
+				ID:         "2037fc3d-be6c-4489-94cf-05518bac709f",
+				Tenant:     tenant,
+				Key:        key,
+				Value:      "val2",
+				ObjectID:   "bar",
+				ObjectType: model.ApplicationLabelableObject,
+			},
+		}
+
+		defWithID := in
+		defWithID.ID = fixUID()
+
+		mockRepository.On("GetByKey", mock.Anything, tenant, key).Return(&ld, nil).Once()
+		mockRepository.On("Update", mock.Anything, defWithID).Return(nil)
+
+		mockLabelRepository.On("ListByKey", context.TODO(), tenant, key).Return(existingLabels, nil).Once()
+
+		ctx := context.TODO()
+		sut := labeldef.NewService(mockRepository, mockLabelRepository, nil)
+		// WHEN
+		actual, err := sut.Update(ctx, in)
+		// THEN
+		require.NoError(t, err)
+		assert.Equal(t, expectedLD, actual)
+	})
+
+	t.Run("returns error when existing label doesn't match new schema", func(t *testing.T) {
+		// GIVEN
+		mockRepository := &automock.Repository{}
+		defer mockRepository.AssertExpectations(t)
+
+		mockLabelRepository := &automock.LabelRepository{}
+		defer mockLabelRepository.AssertExpectations(t)
+
+		newSchema := fixSchema(t, "nonExistingProp", "integer", "desc")
+
+		ld := model.LabelDefinition{
+			ID:     fixUID(),
+			Tenant: tenant,
+			Key:    key,
+			Schema: fixBasicSchema(t),
+		}
+
+		in := model.LabelDefinition{
+			ID:     fixUID(),
+			Key:    key,
+			Tenant: tenant,
+			Schema: newSchema,
+		}
+
+		existingLabels := []*model.Label{
+			{
+				ID:         "b9566e9d-83a2-4091-8c65-7a512b88f89e",
+				Tenant:     tenant,
+				Key:        key,
+				Value:      "val",
+				ObjectID:   "foo",
+				ObjectType: model.RuntimeLabelableObject,
+			},
+			{
+				ID:         "2037fc3d-be6c-4489-94cf-05518bac709f",
+				Tenant:     tenant,
+				Key:        key,
+				Value:      "val2",
+				ObjectID:   "bar",
+				ObjectType: model.ApplicationLabelableObject,
+			},
+		}
+
+		defWithID := in
+		defWithID.ID = fixUID()
+
+		mockRepository.On("GetByKey", mock.Anything, tenant, key).Return(&ld, nil).Once()
+
+		mockLabelRepository.On("ListByKey", context.TODO(), tenant, key).Return(existingLabels, nil).Once()
+
+		ctx := context.TODO()
+		sut := labeldef.NewService(mockRepository, mockLabelRepository, nil)
+		// WHEN
+		_, err := sut.Update(ctx, in)
+		// THEN
+		require.Error(t, err)
+		require.EqualError(t, err, "label with key firstName is not valid against new schema for Runtime with ID foo")
```

Comment:
Instead of calling `fixBasicSchema`, call `fixSchema` with property name  `firstName` It's easier to read.

Reply:
done